### PR TITLE
Makefile: make linter check the commit from upstream master

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary: 

The golangci-lint option `--new-from-rev=HEAD~50` may not be suitable even if the commits number in a PR is less than 50. Because some of the PRs are based on an old commit, and `HEAD~50` does not count the commits from `git merge`. It is possible that an ancient commit is checked(much older than the `HEAD~50` commit of https://github.com/pingcap/tidb/pull/22824).

### What is changed and how it works?

What's Changed: Fetch the latest commit from https://github.com/pingcap/tidb first in `make static-check`.

### Related changes

NA

### Check List 

Tests 

- No code

Side effects

NA

### Release note 

- No release note

### CODE REVIEW

**Review by xhebox — COMMENTED** [review]
BTW, it will find a much older ref, if my local branch is outdated: the `remote/master` ref is so new that it is not in the history of local branch.

```
[xhe@PC tidb]$ git for-each-ref --merged=HEAD --no-contains=HEAD --sort=-committerdate --count=1
c704b9756e20d172da32e95f2c7bec3c40dbaff3 commit refs/remotes/pingcap/stats_cache
# a branch that forked/checkouted from master, sometimes ago
```

File: tools/check/check-latest-ref.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#  -e: exit immediately when a command returning a non-zero exit code.
+#  -u: treat unset variables as an error.
+#  -o pipefail: sets the exit code of a pipeline to that of the rightmost command to exit with a non-zero status,
+#       or to zero if all commands of the pipeline exit successfully.
+set -euo pipefail
+
+latest_ref=$(git for-each-ref --merged=HEAD --no-contains=HEAD --sort=-committerdate --count=1)
+branch=$(echo "$latest_ref" | cut -d' ' -f2)
+
+if [[ ($branch == *"commit"*) && ($branch == *"remotes"*) && ($branch == *"master"*) ]]; then
```
[the remark this answers is not in the document]
Comment (tangenta) [reply tools/check/check-latest-ref.sh:25]: How about __remotes && master__? The CI always pulls the master first.

[the remark this answers is not in the document]
Comment (xhebox) [reply tools/check/check-latest-ref.sh:25]: Same applies to the local env: 

```
[xhe@PC tidb]$ git for-each-ref --merged=HEAD --no-contains=HEAD --sort=-committerdate --count=1
745d20aa64395916060a8f716cdecf9cad9b6957 commit refs/remotes/pingcap/master
```

[the remark this answers is not in the document]
Comment (xhebox) [reply tools/check/check-latest-ref.sh:25]: If you really want to use `HEAD~50` for local dev, this is not reliable.

**Review by AilinKid — COMMENTED** [review]
LGTM

**Review by crazycs520 — APPROVED** [review]
LGTM

**Comment by hi-rustin** [comment]
> /merge

You only need to trigger /merge once, and if the CI test fails, you just re-trigger the test that failed and the bot will merge the PR for you after the CI passes.

**Comment by hi-rustin** [comment]
@ti-chi-bot  update this pr branch

**Comment by hi-rustin** [comment]
@ti-chi-bot update this pr branch

**Pull request merged**